arbor/communication/mpi.hpp

```diff
@@ -48,15 +48,19 @@ struct mpi_traits<T> {  \
     constexpr static bool is_mpi_native_type() { return true; } \
 };
 
-MAKE_TRAITS(double, MPI_DOUBLE)
-MAKE_TRAITS(float,  MPI_FLOAT)
-MAKE_TRAITS(int,    MPI_INT)
-MAKE_TRAITS(long int, MPI_LONG)
-MAKE_TRAITS(char,   MPI_CHAR)
-MAKE_TRAITS(unsigned int, MPI_UNSIGNED)
-MAKE_TRAITS(size_t, MPI_UNSIGNED_LONG)
-static_assert(sizeof(size_t)==sizeof(unsigned long),
-              "size_t and unsigned long are not equivalent");
+MAKE_TRAITS(float,              MPI_FLOAT)
+MAKE_TRAITS(double,             MPI_DOUBLE)
+MAKE_TRAITS(char,               MPI_CHAR)
+MAKE_TRAITS(int,                MPI_INT)
```

Comment:
Should we be explicit about int being signed int for documentary purposes?

Reply:
So now we have traits for types that should never be used? I'm wondering if there's a way to 'break' the types by default, and then fix the size_t type.

Reply:
I think it is safe to assume that people with the fingers in the code at this point won't need the signedness of int to be made explicit — indeed, I think adding in signed just here would make the less confident coder wonder if there is something they are missing elsewhere, and thus increase confusion.

We already had traits for types that don't have an interface in `distributed_context`, but I think that's okay. Having the traits for the basic numeric scalar types is fine IMO. I wouldn't object if `short` and `unsigned short` were added too, for example.

Reply:
Regarding the problem of `size_t`: the maximally portable approach involves a huge amount of type traits guff, and the other portable alternative is to do configure-time checks against the various int types. I'm really, really reluctant to put that work in :)

You can all curse me when we port it to the first system where `size_t`, `uintmax_t`, and `uint128_t` are all the same while `unsigned long long` is 64 bit.

Reply:
I vote for keeping it simple @halfflat.

How about a quick `static_assert` that borks when the highly unlikely scenario that you just described occurs, with a comment attached that explains the situation?

Reply:
@bcumming The appropriate `static_assert` is in the PR :)